### PR TITLE
Additional pipeline to tag the image with latest

### PR DESCRIPTION
**Pull request by zregvart**
~Runs a second pipeline to build and push an image tagged with the git commit id.~
Runs a second pipeline that waits 1 min for the first pipeline to push the image and then copies the image with the `latest` tag.

### CODE REVIEW

**Comment by lcarva** [comment]
I don't think we want to build the image twice. This will not, for example, retain the digest of image currently tagged as latest. That image will still get cleaned up.

**Comment by zregvart** [comment]
I have no idea how to test this and if this works at all.

**Comment by lcarva** [comment]
Once you're happy with it, feel free to merge it. We may have to do a couple of iterations until we get it right. That's ok.

File: .tekton/tag-latest.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      - name: tag-with-latest
+        params:
+          - name: revision
+            value: '$(params.revision)'
+        taskSpec:
+          params:
+            - name: revision
+          steps:
+            - name: tag-with-latest
+              image: bash
```
[the remark this answers is not in the document]
Comment (zregvart) [reply .tekton/tag-latest.yaml:29]: I was testing with `registry.access.redhat.com/ubi9/skopeo`, forgot to update here

File: .tekton/tag-latest.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                #!/usr/bin/env bash
+                SRC_REF=quay.io/redhat-appstudio/ec-golden-image:$(params.revision)
+                TARGET_REF=quay.io/redhat-appstudio/ec-golden-image:latest
+
+                retries=0
+                while ! skopeo inspect docker://${SRC_REF} >/dev/null 2>&1; do
+                  (( retries++ ))
+                  if (( retries > 20 )); then
+                    echo
+                    echo ${SRC_REF} has not been pushed within 1 min
```
[the remark this answers is not in the document]
Comment (zregvart) [reply .tekton/tag-latest.yaml:40]: Yeah, it takes about 2min now

**Comment by zregvart** [comment]
> Once you're happy with it, feel free to merge it. We may have to do a couple of iterations until we get it right. That's ok.

Yeah, I think I need to test in production

**Comment by zregvart** [comment]
This never happens, worked on first try :)

```
$ cosign tree quay.io/redhat-appstudio/ec-golden-image:latest
📦 Supply Chain Security Related artifacts for an image: quay.io/redhat-appstudio/ec-golden-image:latest
└── 🔐 Signatures for an image tag: quay.io/redhat-appstudio/ec-golden-image:sha256-6858e6cb3ccf9caadf79c30f679a26f2daeb84b999b77561a3e9000d9b983441.sig
   └── 🍒 sha256:fa758aaf821d43ea323ed5d9bb937ef5bfce8e84f2101eb872b19627f90f6a24
└── 📦 SBOMs for an image tag: quay.io/redhat-appstudio/ec-golden-image:sha256-6858e6cb3ccf9caadf79c30f679a26f2daeb84b999b77561a3e9000d9b983441.sbom
   └── 🍒 sha256:963b025cf1988ec6945ec21e830bca7792aa20f34aa9f4287e40fafb404fdca7
└── 💾 Attestations for an image tag: quay.io/redhat-appstudio/ec-golden-image:sha256-6858e6cb3ccf9caadf79c30f679a26f2daeb84b999b77561a3e9000d9b983441.att
   └── 🍒 sha256:b0510cb023c91c90df16595915ec1180039c94880cdcc278b4cc9ff68af4d8c8

$ cosign tree quay.io/redhat-appstudio/ec-golden-image:c2f9e02e272ff8da32133ebcff454560c9dd0e38
📦 Supply Chain Security Related artifacts for an image: quay.io/redhat-appstudio/ec-golden-image:c2f9e02e272ff8da32133ebcff454560c9dd0e38
└── 💾 Attestations for an image tag: quay.io/redhat-appstudio/ec-golden-image:sha256-6858e6cb3ccf9caadf79c30f679a26f2daeb84b999b77561a3e9000d9b983441.att
   ├── 🍒 sha256:b0510cb023c91c90df16595915ec1180039c94880cdcc278b4cc9ff68af4d8c8
   └── 🍒 sha256:399147f4350cd7a6e1f0cd1589f7a48fd68ab494f041317c27389d2a5ecc2f28
└── 🔐 Signatures for an image tag: quay.io/redhat-appstudio/ec-golden-image:sha256-6858e6cb3ccf9caadf79c30f679a26f2daeb84b999b77561a3e9000d9b983441.sig
   └── 🍒 sha256:fa758aaf821d43ea323ed5d9bb937ef5bfce8e84f2101eb872b19627f90f6a24
└── 📦 SBOMs for an image tag: quay.io/redhat-appstudio/ec-golden-image:sha256-6858e6cb3ccf9caadf79c30f679a26f2daeb84b999b77561a3e9000d9b983441.sbom
   └── 🍒 sha256:963b025cf1988ec6945ec21e830bca7792aa20f34aa9f4287e40fafb404fdca7
```

**Comment by lcarva** [comment]
Yay! That is truly amazing.

It occurred to me, how would a Stonesoup user do this? I think the goal is to have them modify the pipeline itself, which would be easier with an expanded pipeline definition (which is the plan afaik).

**Comment by zregvart** [comment]
Yeah, I did not want to expand the pipeline here, if a new test is added we would miss it.

**Pull request merged**